docs/posts/into-snowflake/index.qmd

```diff
@@ -0,0 +1,127 @@
+---
+title: "Snow IO: loading data from other DBs into Snowflake"
+author: "Phillip Cloud"
+error: false
+date: "2024-02-22"
+categories:
+  - blog
+  - snowflake
+  - io
+  - productivity
+---
+
+## Recap
+
+We've [blogged about Snowflake IO before](../snowflake-io/index.qmd), in the
+context of getting local files into Snowflake as fast as possible.
+
+In this post, we'll show how query another system and insert query results into
+Snowflake chunk by chunk.
+
+## Setup
+
+### Connect to your non-Snowflake system
+
+We'll connect to a postgres database running locally in a container. You
+should be able to swap in your own connection details as needed.
+
+```{python}
+from ibis.interactive import *  # <1>
+
+pg_con = ibis.connect("postgres://postgres:postgres@localhost/postgres")
+```
+
+1. Import Ibis for maximum productivity in interactive analysis.
+
+We'll use a test dataset that contains some baseball batting statistics.
+
+Ibis provides that example data, so we can dump that into postgres.
+
+
+```{python}
+pg_batting = pg_con.create_table(
+    "batting",
+    ibis.examples.Batting.fetch().to_pandas(),  # <1>
+    temp=True,  # <2>
+)
+```
+
+1. Yep, I'm using pandas here!
+2. Use a temporary table to avoid cluttering up the database.
+
+### Connect to Snowflake
+
+```{python}
+import os
+
+snow_con = ibis.connect(os.environ["SNOWFLAKE_URL"])  # <1>
+```
+
+1. Set the `SNOWFLAKE_URL` environment variable to your Snowflake connection
+   string, which will look like `snowflake://user:pass@account/database/schema?warehouse=my_warehouse`.
+
+## Profit
+
+### Construct an Ibis expression from the postgres data
+
+Let's build an Ibis expression based on the `batting` table in our postgres database.
+
+```{python}
+pg_batting
+```
+
+Let's compute the average RBI per year per team.
+
+```{python}
+pg_expr = pg_batting.group_by(("year_id", "team_id")).agg(avg_rbi=_.rbi.mean())
+pg_expr
+```
+
+Let's show how many rows we have in the result.
+
+```{python}
+pg_expr.count()
+```
+
+### Insert the computed results into Snowflake
+
+Because all of our backends implement the `to_pyarrow()` method, we can
+get data out of another system and into Snowflake with a few lines of code.
+
+First we'll create a table in Snowflake to hold the data. Ibis helps here by
+providing an API to access the schema from the **postgres**-based expression,
+so you don't need to worry about translating between the postgres type system
+and the Snowflake type system.
```

Comment:
Maybe here we spell out more directly that "...Ibis translates from the postgres type system to the Snowflake type system so you don't have to"?

Reply:
Done